linfa-bayes/src/gaussian_nb.rs

```diff
@@ -97,32 +83,29 @@ where
         let mut unique_classes = dataset.targets.labels();
         unique_classes.sort_unstable();
 
-        let mut model = GaussianNb::unfitted();
-        if self.priors_provided {
-            model.priors = self.priors.clone();
-        }
+        let mut model: Option<GaussianNb<_>> = None;
 
         // We train the model
-        model = self.fit_with(
-            model,
-            dataset.records.view(),
-            &dataset.targets,
-            &unique_classes,
-        )?;
-
-        Ok(model)
+        model = self.fit_with(model, dataset)?;
+
+        Ok(model.unwrap())
     }
 }
 
-impl<A: Float> GaussianNbParams<A> {
+impl<A, L> IncrementalFit<'_, ArrayView2<'_, A>, L, BayesError> for GaussianNbParams
+where
+    A: Float + PartialEq + PartialOrd,
```

Comment:
you can just bound this in `linfa::Float`, it wasn't done to this point because no other algorithm used partial equality/ordering